CMakeLists.txt

```diff
@@ -0,0 +1,294 @@
+  GIT_TAG        1.0.6
+)
+FetchContent_MakeAvailable(bls)
+
+set(CMAKE_C_FLAGS_RELEASE_INIT)
+set(CMAKE_C_FLAGS_DEBUG_INIT)
+set(CMAKE_CXX_FLAGS_RELEASE_INIT)
+set(CMAKE_CXX_FLAGS_DEBUG_INIT)
+
+
+set(c_opts)
+set(link_opts)
+# set(cxx_opts)
+
+set(release_c_opts)
+set(debug_c_opts)
+set(dev_c_opts)
+
+set(release_link_opts)
+set(debug_link_opts)
+
+# Config
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+
+    # MSVC
+    set(c_opts
+        /std:c++17
+        /Zc:__cplusplus
+        /MP
+        /Zi
+        /EHsc-
+        # /Wall
+        /W3
+        /WX
+        /FIpch.h 
+        /wd4068
+        /wd4464
+        /wd4668
+        /wd4820
+        /wd4514
+        /wd4626
+        /wd5027
+        /DUNICODE=1
+        /DWIN32_LEAN_AND_MEAN=1
+        /DNOMINMAX=1
+        /D_CRT_SECURE_NO_WARNINGS=1
+        /D_HAS_EXCEPTIONS=0
+        ${c_opts})
+    
+    set(link_opts 
+        /SUBSYSTEM:CONSOLE
+        ${link_opts})
+
+    set(release_c_opts 
+        /MT
+        /O2
+        /Oi
+        /Gy
+        /GL
+        /DNDEBUG=1
+        /DNDEBUG=1
+        ${release_c_opts})
+
+    set(debug_c_opts 
+        /MTd
+        /Od
+        ${debug_c_opts})
+
+    set(dev_c_opts 
+        ${dev_c_opts})
+
+    
+    set(release_link_opts 
+        /DEBUG:FULL
+        /OPT:REF,IFC,LBR
+        ${release_link_opts})
+
+    set(debug_link_opts
+        /DEBUG:FASTLINK
+        /OPT:NOREF,NOICF,NOLBR
+        /INCREMENTAL
+        ${debug_link_opts})
+
+    # Dependency config
+    target_compile_options(bls     PRIVATE /MP)
+    target_compile_options(relic_s PRIVATE /MP)
+    target_compile_options(sodium  PRIVATE /MP)
+
+    target_compile_options(bls     PRIVATE $<$<CONFIG:release>:/MT>)
+    target_compile_options(relic_s PRIVATE $<$<CONFIG:release>:/MT>)
+    target_compile_options(sodium  PRIVATE $<$<CONFIG:release>:/MT>)
+
+    target_compile_options(bls     PRIVATE $<$<CONFIG:debug>:/MTd>)
+    target_compile_options(relic_s PRIVATE $<$<CONFIG:debug>:/MTd>)
+    target_compile_options(sodium  PRIVATE $<$<CONFIG:debug>:/MTd>)
+    
+else()
+
+    # *Nix
+    set(CMAKE_C_FLAGS --include=pch.h -Wall -Wno-comment -g ${CMAKE_C_FLAGS})
+    set(CMAKE_CXX_FLAGS -std=c++17 ${CMAKE_CXX_FLAGS})
+
+    set(release_c_opts 
+        -O3 -flto
+        ${release_c_opts})
+
+    set(debug_c_opts 
+        -O0
+        ${debug_c_opts})
+
+    set(dev_c_opts 
+        ${dev_c_opts})
+
+    # GCC
+    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    
+        set(CMAKE_C_FLAGS -fmax-errors=5 ${CMAKE_C_FLAGS})
+
+    # Clang
+    elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+        
+        set(CMAKE_C_FLAGS -ferror-limit=5 -fdeclspec -fno-exceptions -Wunknown-pragmas ${CMAKE_C_FLAGS})
+
+    endif()
+
+endif()
+
+
+# Main Sources
+file(GLOB_RECURSE bb_sources
+    RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
+    CONFIGURE_DEPENDS
+    LIST_DIRECTORIES false
+    src/*.cpp
+    src/*.c
+)
+
+# Headers
+file(GLOB_RECURSE bb_headers
```

Comment:
I would make the list of source files explicit to avoid using GLOB as per CMake recommendation: https://cmake.org/cmake/help/latest/command/file.html?highlight=Note#filesystem